awe_languagetool/languagetoolClient.py

```diff
@@ -53,9 +51,43 @@ class languagetoolClient:
                                + 1) if not chr(i).isprintable()
     }
 
-    def __init__(self, port=8081):
+    def __init__(self, port=8081, host='localhost', server_url=None):
+        """
+        Handles initializing files and connections for the LanguageTool client
+        to use.
+
+        Parameters:
+        -----------
+        port : int, optional
+            The port on which the server is running. Defaults to 8081 if not provided.
+
+        host : str, optional
+            The hostname or IP address of the server. Defaults to 'localhost' if not provided.
+
+        server_url : str, optional
+            The full URL of the server. If provided, this URL is used as `self.server_url`.
+            If not provided, `self.server_url` is constructed using the specified `host` and `port`.
 
-        self.port = port
+        Notes:
+        ------
+        If `server_url` is specified, it overrides any values set by `host` and `port`.
+        """
+        self.server_url = server_url or f'http://{host}:{port}'
+
+        # Try connecting to the server and raise a Runtime error if we can't reach it
+        server_not_available_error = f'LanguageTool Server was not found running on {self.server_url}.\n'\
+            'Please make sure to run the LanguageTool Server before using LanguageTool Client. Run:\n'\
+            '`python -m awe_languagetool.languagetoolServer`\n'\
+            'OR with a configuration file\n'\
+            '`python -m awe_languagetool.languagetoolServer --config /path/to/config.cfg`\n'\
+            'Ensure you are using the correct url.'\
+
+        try:
+            resp = requests.get(f'{self.server_url}/v2/check', params={'text': 'test', 'language': 'en-US'})
+            if resp.status_code != 200:
+                raise RuntimeError(server_not_available_error)
```

Comment:
This should probably mention we got a non-200 response from whatever server we are trying to ping.